codegen/templates.go

```diff
@@ -1,5 +1,258 @@
+		UpperCaseDifferentiator: strings.Title(strings.ToLower(lastEndpointField)),
+		LowerCaseDifferentiator: strings.ToLower(lastEndpointField),
+		Parameters:              parameters,
+		SupportsRead:            endpointInfo.Get != nil,
+		SupportsWrite:           endpointInfo.Post != nil,
+		SupportsDelete:          endpointInfo.Delete != nil,
+	}
+	if err := t.Validate(); err != nil {
+		return nil, err
+	}
+	return t, nil
+}
+
+// collectParameters walks a PathItem and looks for all the parameters
+// described. Some are at the top level of the path, indicating they are
+// path parameters. Others are only in the post body.
+func collectParameters(endpointInfo *framework.OASPathItem) []*templatableParam {
+	var result []*templatableParam
+	for _, param := range endpointInfo.Parameters {
+		result = append(result, toTemplatableParam(param, true))
+	}
+	if endpointInfo.Post == nil {
+		return result
+	}
+	if endpointInfo.Post.RequestBody == nil {
+		return result
+	}
+	if endpointInfo.Post.RequestBody.Content == nil {
+		return result
+	}
+	for _, mediaTypeObject := range endpointInfo.Post.RequestBody.Content {
+		if mediaTypeObject.Schema == nil {
+			continue
+		}
+		if mediaTypeObject.Schema.Properties == nil {
+			continue
+		}
+		for paramName, schema := range mediaTypeObject.Schema.Properties {
+			param := framework.OASParameter{
+				Name:        paramName,
+				Description: schema.Description,
+				In:          "post",
+				Schema:      schema,
+			}
+			result = append(result, toTemplatableParam(param, false))
+		}
+	}
+	return result
+}
+
+// templatableParam mainly just reuses the OASParameter,
+// but adds on a IsPathParam bool.
+type templatableParam struct {
+	*framework.OASParameter
+	IsPathParam bool
+}
+
+func toTemplatableParam(param framework.OASParameter, isPathParameter bool) *templatableParam {
+	ptrToParam := &param
+	if ptrToParam.Schema == nil {
+		// Always populate schema and display attributes so later it'll be easier
+		// to check if they're sensitive by iterating over them.
+		ptrToParam.Schema = &framework.OASSchema{}
+	}
+	if ptrToParam.Schema.DisplayAttrs == nil {
+		ptrToParam.Schema.DisplayAttrs = &framework.DisplayAttributes{}
+	}
+	return &templatableParam{
+		OASParameter: ptrToParam,
+		IsPathParam:  isPathParameter,
+	}
+}
+
+// templatableEndpoint is a convenience struct that plays nicely with Go's
+// template package. It is used to keep as much logic as possible in Go
+// rather than in Go's templating language, because most folks are more
+// familiar with Go.
+type templatableEndpoint struct {
+	Endpoint                string
+	DirName                 string
+	UpperCaseDifferentiator string
+	LowerCaseDifferentiator string
+	Parameters              []*templatableParam
+	SupportsRead            bool
+	SupportsWrite           bool
+	SupportsDelete          bool
+}
+
+func (e *templatableEndpoint) Validate() error {
+	if e == nil {
+		return fmt.Errorf("endpoint is nil")
```

Comment:
We can't on this line specifically because if we continue on, the code tries to access attributes on a nil pointer and panics - but definitely the answer is yes for the _rest_ of the errors in this func. I'll add multierr!

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/746/commits/c3afd186cc031e2284d730fcd4969aed2438e88a